lib/pharos/phases/validate_host.rb

```diff
@@ -46,6 +48,11 @@ def validate_unique_hostnames
         raise Pharos::InvalidHostError, "Duplicate hostname #{@host.hostname} for hosts #{duplicates.map(&:address).join(',')}"
       end
 
+      def validate_localhost_resolve
+        localhost_ip = @ssh.exec!("ping -c 1 localhost | grep -Eo -m 1 '[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}';")
```

Comment:
Could be `getent ahostsv4 localhost | grep STREAM | head -n 1 | cut -d ' ' -f1` in case there's an old version of grep

Reply:
Tested with CentOS7 which probably is the oldest.